apps/src/signUpFlow/FinishTeacherAccount.tsx

```diff
@@ -0,0 +1,107 @@
+import React, {useState} from 'react';
+
+import {Button, buttonColors} from '@cdo/apps/componentLibrary/button';
+import Checkbox from '@cdo/apps/componentLibrary/checkbox/Checkbox';
+import TextField from '@cdo/apps/componentLibrary/textField/TextField';
+import {
+  Heading2,
+  BodyTwoText,
+  BodyThreeText,
+} from '@cdo/apps/componentLibrary/typography';
+import locale from '@cdo/apps/signup/locale';
+import SchoolDataInputs from '@cdo/apps/templates/SchoolDataInputs';
+
+import {
+  USER_NAME_SESSION_KEY,
+  EMAIL_OPT_IN_SESSION_KEY,
+} from './signUpFlowConstants';
+
+import style from './finish-account.module.scss';
+
+const FinishTeacherAccount: React.FunctionComponent<{
+  usIp: boolean;
+}> = ({usIp}) => {
+  const [name, setName] = useState('');
+  const [emailOptInChecked, setEmailOptInChecked] = useState(false);
+
+  const onNameChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    const newName = e.target.value;
+    setName(newName);
+    sessionStorage.setItem(USER_NAME_SESSION_KEY, newName);
+  };
+
+  const onEmailOptInChange = (): void => {
+    const newOptInCheckedChoice = !emailOptInChecked;
+    setEmailOptInChecked(newOptInCheckedChoice);
+    sessionStorage.setItem(
+      EMAIL_OPT_IN_SESSION_KEY,
+      `${newOptInCheckedChoice}`
+    );
+  };
+
+  return (
+    <div className={style.finishTeacherAccountContainer}>
+      <div className={style.headerTextContainer}>
+        <Heading2>{locale.finish_creating_teacher_account()}</Heading2>
+        <BodyTwoText>{locale.tailor_experience()}</BodyTwoText>
+      </div>
+      <div className={style.inputContainer}>
+        <BodyTwoText visualAppearance={'heading-xs'}>
+          {locale.what_do_you_want_to_be_called()}
+        </BodyTwoText>
+        <TextField
+          name="userName"
+          onChange={onNameChange}
+          value={name}
+          className={style.nameInput}
+          placeholder={'Dr. Joanna'}
```

Comment:
I figured since this was a name we wouldn't want to translate it, but it does also contain "Dr." in it so I'm unsure. Thoughts on this?

Reply:
This could come as a follow-up, but I'd translate this. "Dr. Joanna" isn't referring to a real person, just the general idea of a placeholder name

Reply:
But thinking on this, is this an appropriate placeholder for "userName"? It's more of a display name, no? (This might be something to address in this PR)

Reply:
@bethanyaconnor can you clarify about the username aspect? Since this field was intended for display name, I think it makes sense to use a placeholder that indicates this is where they enter what students are seeing and not necessarily their full name and/or a username like user123. I think in the student flow, Dani suggested something like "Coder" or "student," maybe we can do something similar here "Mr/s. Coder?"

Reply:
`username` is a property of users in our database that has specific requirements. It is not what is shown on the top right of the page. For example, the username for my fake teacher account is `bethany_teacher369`.

So what I'm getting at is that I'd like to see this input renamed to `displayName` or something similar, if possible!

Reply:
Sorry for not being clear -- I didn't connect this to the screenshots in the description and thought we were letting users choose their username on sign u 🤦‍♀️

Reply:
Final decision was to use `displayName` over `userName` and to have the placeholder be `Ms. Coder` and translate it.